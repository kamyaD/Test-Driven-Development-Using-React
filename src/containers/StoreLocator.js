import React, { Component } from 'react';
import Header from '../components/Header';
import Button from '../components/Button';
import Map from '../components/Map';
import mapChooser from '../containers/mapChooser';
import Axios from 'axios';


class StoreLocator extends Component {
    constructor(props) {
        super(props);
        this.state = {
            currentMap: 'none.png',
            Markets: []
        }
        this.chooseMap = this.chooseMap.bind(this);
    }
    componentDidMount() {
        this.getMarkets();

    }
    getMarkets = async () => {
        let response = await Axios.get('http://localhost:3000/data/Markets.json')
        this.setState({
            Markets: response.data.Markets
        })
    }
    chooseMap(e) {
        this.setState({ currentMap: mapChooser(e.target.value) });

    }
    render() {
        let locationButtons = this.state.Markets.map((market, id) => { return (<Button handleClick={this.chooseMap} key={id} location={market.Location} />) }
        )
        return (
            <div>
                <Header />
                <div>
                    {locationButtons}
                </div>
                <Map imagename={this.state.currentMap} location={this.props.location} />
            </div>

        );

    }
}

export default StoreLocator;