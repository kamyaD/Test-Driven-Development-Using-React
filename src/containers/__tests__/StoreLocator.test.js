import React from 'react';
import StoreLocator from '../StoreLocator';
import { shallow } from 'enzyme';
import axios from 'axios';
import renderer from 'react-test-renderer'
// import Button from '../../components/Button';

describe("StoreLocator", function () {
  let mountedStoreLocator;
  beforeEach(() => {
    mountedStoreLocator = shallow(<StoreLocator />);
  })

  it('renders correctly', ()=>{
    const tree = renderer.create(<StoreLocator/>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders without crashing', () => {
    let mountedStoreLocator = shallow(<StoreLocator />);

  });

  it('renders a header', () => {
    const headers = mountedStoreLocator.find('Header');
    expect(headers.length).toBe(1);

  });
  it('renders three buttons', () => {
    const Buttons = mountedStoreLocator.find('Button');
    expect(Buttons.length).toBe(3);

  });

  it('renders a map ', () => {
    const map = mountedStoreLocator.find('Map');
    expect(map.length).toBe(1);

  });

  it('calls axios.get in #componentDidMount', () => {
    return mountedStoreLocator.instance().componentDidMount().then(() => {
      expect(axios.get).toHaveBeenCalled();
    })
  });

  it('calls axios.get with correct url', () => {
    return mountedStoreLocator.instance().componentDidMount().then(() => {
      expect(axios.get).toHaveBeenCalledWith('http://localhost:3000/data/Markets.json');
    })
  });

  it('updates state with data from api', () => {
    return mountedStoreLocator.instance().componentDidMount().then(() => {
      expect(mountedStoreLocator.state()).toHaveProperty('Markets',
        [{
          "Location": "test Location",
          "address": "test address"

        }]
      );
    })

  });


});

describe('choose map', () => {
  it('updates this.state.currentMap using the location passed to it', () => {
    let mountedStoreLocator = shallow(<StoreLocator />);
    let mockEvent = { target: { value: 'testland' } };
    mountedStoreLocator.instance().chooseMap(mockEvent);
    expect(mountedStoreLocator.instance().state.currentMap).toBe('testland.jpg');
  })

});

