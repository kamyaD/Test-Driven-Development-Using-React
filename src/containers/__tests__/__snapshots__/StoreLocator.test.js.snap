// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StoreLocator renders correctly 1`] = `
<div>
  <div
    className="Header"
  >
    <img
      alt="Kit mikai"
      src="images/wired-brain-logo.png"
      width="225px"
    />
  </div>
  <div />
  <div
    className="MapBox"
  >
    <img
      src="images/none.png"
    />
  </div>
</div>
`;
